src/python/dxpy/scripts/dx.py

```diff
@@ -3741,6 +3744,7 @@ def register_parser(parser, subparsers_action=None, categories=('other', ), add_
 parser_upload.add_argument('--wait', help='Wait until the file has finished closing', action='store_true')
 parser_upload.add_argument('--no-progress', help='Do not show a progress bar', dest='show_progress',
                            action='store_false', default=sys.stderr.isatty())
+parser_upload.add_argument('--buffer-size', help='Set the write buffer size', dest='write_buffer_size')
```

Comment:
Mention that this should be specified in units of bytes? It would be nice at some point in the future to allow specifying these with a suffix, like ``--buffer-size=16M`.